gapic/templates/%namespace/%name_%version/%sub/services/%service/client.py.j2

```diff
@@ -170,24 +164,27 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
                 raise ValueError('When providing a transport instance, '
                                  'provide its credentials directly.')
             self._transport = transport
-        elif transport is not None or (
-            client_options.api_endpoint == self.DEFAULT_ENDPOINT
+        elif client_options is None or (
+            client_options.api_endpoint == None
             and client_options.client_cert_source is None
         ):
-            # Don't trigger mTLS.
+            # Don't trigger mTLS if we get an empty ClientOptions.
             Transport = type(self).get_transport_class(transport)
             self._transport = Transport(
-                credentials=credentials, host=client_options.api_endpoint
+                credentials=credentials, host=self.DEFAULT_ENDPOINT
             )
         else:
-            # Trigger mTLS. If the user overrides endpoint, use it as the mTLS
-            # endpoint, otherwise use the default mTLS endpoint.
-            option_endpoint = client_options.api_endpoint
-            api_mtls_endpoint = self.DEFAULT_MTLS_ENDPOINT if option_endpoint == self.DEFAULT_ENDPOINT else option_endpoint
+            # We have a non-empty ClientOptions, trigger mTLS. If the user
+            # doesn't provide endpoint, use the default mTLS endpoint.
+            if client_options.api_endpoint:
+                api_endpoint = api_mtls_endpoint = client_options.api_endpoint
+            else:
+                api_endpoint = self.DEFAULT_ENDPOINT
+                api_mtls_endpoint = self.DEFAULT_MTLS_ENDPOINT
```

Comment:
Judgement call on your part, but what do you think about
```python
opt_endpoint = client_options.api_endpoint
endpoint, mtls_endpoint = (opt_endpint, opt_endpoint) if opt_endpoint else (self.DEFAULT_ENDPOINT, self.DEFAULT_MTLS_ENDPOINT)
```

I generally prefer conditional assignment in python, when it isn't too complicated, because it eliminates undefined reference errors; I also find it's often easier to reason about how values flow through code than logic branches.

Reply:
logic is changed in the new code. Sure, I will use conditional assignment when simple and possible.